Capabilities 2020 chapter

christianliebel opened the pull request:
Closes #912

- [x] Links/references to sources by different vendors
- [x] At least one link to the PWA chapter and how this differentiates from that and why it's a separate chapter.
- [x] absolute numbers for charts?
- [x] no SQL files for all parameters
- [x] figure descriptions
- [x] quote
- [x] 3 interesting stats
- [x] Author bio

rviscomi commented:
Thanks @christianliebel! Am I correct to assume that this is the first draft and it hasn't been reviewed by the rest of the team yet? If that's the case, I would suggest closing this PR and iterating on comments/feedback in the [chapter doc](https://docs.google.com/document/d/1rPyoVaWmKUHvB7HGhtQpapAT4W9lF1wzF_hJHRoXou8/edit#heading=h.l58oy8wsputh) instead. If you and your reviewers are comfortable doing it via GitHub PR, you can try that, but I've found that chapters have had an easier time using Docs for that.

rviscomi commented:
Oh! I see you've done the review in your own GitHub repo instead: https://github.com/christianliebel/web-almanac-2020-capabilities/commits/master/README.md

In that case, I think you're right that this is ready for editing and pre-publication. 🚀

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,443 @@
+images as additional data formats. Due to privacy reasons, reading from the clipboard always
+requires the user's consent.
+
+(IMAGE: Async Clipboard API)
```
For any visual or tabular figures to be embedded in the chapter, please refer to the [Figures Guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) in the wiki for info on the expected markdown syntax to use and how to add metadata.

christianliebel replied:
Done, left the description at "TODO" until we decide which data to choose

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,443 @@
+# Capabilities
```
The title is auto-generated based on YAML metadata, which also includes info for authors, reviewers, analysts, etc.

Please see https://raw.githubusercontent.com/HTTPArchive/almanac.httparchive.org/main/src/content/en/2019/performance.md for an example of the metadata to include.

cc @bazzadp

christianliebel replied:
Almost done, I just need the ID of the new Discourse thread for this chapter.

bazzadp replied:
Let’s use the same id as last year for now.

christianliebel replied:
Capabilities is a new chapter, so there's no thread from last year.

bazzadp replied:
Ah silly me. Go with 1981 for now which is the 2020 announcement.

rviscomi reviewed (changes requested):
A couple of housekeeping things before doing a thorough review.

christianliebel commented:
> In that case, I think you're right that this is ready for editing and pre-publication. 🚀

@rviscomi Yes, this is actually my final draft and it has already been reviewed by @tomayac. However, we need to wait for @logicalphase's approval. In the meantime, I will add the missing images etc.

bazzadp commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,459 @@
+
+## Introduction
+
+The Capabilities Project, informally also known as Project Fugu, is a cross-company effort by
+Google, Microsoft, and Intel to close the gap between native applications and web apps. To do so,
+the Chromium contributors implement new APIs exposing capabilities of the operating system to the
+web, while maintaining user security, privacy, and trust. These capabilities include, but are not
+limited to:
```
Paragraphs should be written on one line as makes it easier to amend and also translate without having to reorder the lines completely and show more change than really is.

```suggestion
The Capabilities Project, informally also known as Project Fugu, is a cross-company effort by Google, Microsoft, and Intel to close the gap between native applications and web apps. To do so, the Chromium contributors implement new APIs exposing capabilities of the operating system to the web, while maintaining user security, privacy, and trust. These capabilities include, but are not limited to:
```

bazzadp commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,459 @@
+
+The Async Clipboard API is comparatively new, so its usage is currently rather low. In March 2020,
+Safari added support for the Async Clipboard API in Safari 13.1. Over the course of 2020, the usage
+of the `read()` API was growing. In October 2020, the API was called during 0.0003 percent of all
```
In general we do encourage percentage stats rather than absolute stats, however I wonder should we make an exception here? 0.0003 percent of sites is a difficult to comprehend number (except that it's really small!) so should we quantify this with an actual number:

```suggestion
of the `read()` API was growing. In October 2020, the API was called during 0.0003 percent of all page loads in Google Chrome (or approximately 20 pages).
```

Or something like that. What do you think @rviscomi ?

rviscomi replied:
If most of the stats are this small, I'd recommend a paragraph to set expectations with readers like:

> Many of the capabilities discussed in this chapter are still very new. So, unlike most chapters, stats will be presented as the absolute numbers of pages rather than relative percentages.

christianliebel replied:
@rviscomi Yes, most of the stats are this small. We need to check if we have absolute numbers for the Chrome Platform Status metrics available though.

rviscomi replied:
Sorry it's taken me so long to understand that these numbers are from Chrome Status and not HTTP Archive. That being the case, only relative stats are available.

For the HTTP Archive-based stats like StorageManager API :: Estimate, we should switch over from percent of pages to number of pages for consistency with the other HA-based stats that have even smaller adoption and need to use absolute page counts. I think it's ok for HA and Chrome-based stats to be different in terms of % of **page views** and number of **pages**.

bazzadp reviewed (changes requested):
Hi @christianliebel ,

Interesting read!

Some general feedback for now:

Can you merge the latest main branch into this branch (or let me know if you're cool for me to do that). I just pushed a fix in #1384 which will fix the build here which will allow for the ultimate HTML to be linted for issues.

Where are the queries @tomayac ?

It reads very Chrome/Google heavy - with nearly all references being to Google Chrome and web.dev (run by Google). I know Google is driving a lot of this, and you all work for them, but as an outsider it could read a little one-sided because of that. I note you say it's a cross community project with Microsoft and Intel - are there any more links or references we could put in from them to address this somewhat? Or some Safari links for when you've mentioned them?

I think there should be at least one link to PWA chapter and how this differentiates from that and why it's a separate chapter.

bazzadp commented:
I've deployed a test version of the chapter here so you can see what it looks like: https://20201025t062109-dot-webalmanac.uk.r.appspot.com/en/2020/capabilities. Looking nice!

A few more comments from me:

- We're looking for more meaningful query names. So `durable-storage-estimate-usage.sql` rather than `13_01.sql` for example.
- Looks like you missed the `sql_file` parameter from a few queries. This is not used yet but we hope to [add the ability to link directly to SQL](https://github.com/HTTPArchive/almanac.httparchive.org/issues/1138) like we do on the [HTTP Archive reports site](https://httparchive.org/reports/project-fugu) so would be good to have them if we can for when we add this.
- Thanks for merging the newlines. Personally I would take it one step further and have all paragraphs on one line but not a big deal.
- Some of the graph's are a little hard to read due to long, overlapping labels. Might be resolved if we move to absolute numbers instead of percentages as mentioned previously? Or some other solution.
- Can we change the QUIC label on the graph to QuicTransport to differentiate from QUIC that will be discussed in the HTTP/2 chapter?

Authors get a bio so we'll need a line or two from you @christianliebel about who you are. See [2019 examples](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/templates/en/2019/base.html#L180) below:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/e768b875bbc270d3e510897b49416fd746622039/src/templates/en/2019/base.html#L180-L185

Don't have the place to put this yet (it depends on merging #1383 ) so just something to think about for now. Or comment here and we'll add it when ready.

Similarly we need a [quote and 3 interesting stats](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/templates/en/2019/featured_chapters.html) from this chapter for the random Featured Quote section of the [home page](https://almanac.httparchive.org/en/2020/). Again not quite ready for this just yet so only a heads up for now.

christianliebel commented:
> Can you merge the latest main branch into this branch (or let me know if you're cool for me to do that). I just pushed a fix in #1384 which will fix the build here which will allow for the ultimate HTML to be linted for issues.

@bazzadp Done.

> Where are the queries @tomayac ?

I added them to the repository in the meantime. All queries can also be found here: https://httparchive.org/reports/project-fugu

> It reads very Chrome/Google heavy - with nearly all references being to Google Chrome and web.dev (run by Google). I know Google is driving a lot of this, and you all work for them, but as an outsider it could read a little one-sided because of that. I note you say it's a cross community project with Microsoft and Intel - are there any more links or references we could put in from them to address this somewhat? Or some Safari links for when you've mentioned them?

I also don't work for Google, but I see your point. The problem there is that, to my knowledge, there's barely any (primary) sources by other vendors with the rare exception of articles like this: https://webkit.org/blog/10855/async-clipboard-api/ But I can definitely go and find some more resources.

TODO @christianliebel / maybe @tomayac is aware of other resources?

> I think there should be at least one link to PWA chapter and how this differentiates from that and why it's a separate chapter.

Sure. TODO @christianliebel 

> We're looking for more meaningful query names. So durable-storage-estimate-usage.sql rather than 13_01.sql for example.

Done.

> Looks like you missed the sql_file parameter from a few queries. This is not used yet but we hope to add the ability to link directly to SQL like we do on the HTTP Archive reports site so would be good to have them if we can for when we add this.

The figures without SQL queries are not drawn from HTTP Archive data, but from the Chrome Platform Status metrics instead. I'm not sure if 

> Thanks for merging the newlines. Personally I would take it one step further and have all paragraphs on one line but not a big deal.

Changed.

> Some of the graph's are a little hard to read due to long, overlapping labels. Might be resolved if we move to absolute numbers instead of percentages as mentioned previously? Or some other solution.

Sure, we could work with absolute numbers (at least regarding the HTTP Archive data). I'm not sure if we can get access to the number of absolute pages for the Chrome Platform Status metrics. @tomayac: Do you know more about that? 

> Can we change the QUIC label on the graph to QuicTransport to differentiate from QUIC that will be discussed in the HTTP/2 chapter?

Sure. TODO @christianliebel 

> Authors get a bio so we'll need a line or two from you @christianliebel about who you are. See 2019 examples below:

I added my bio to the the top comment of this PR.

> Similarly we need a quote and 3 interesting stats from this chapter for the random Featured Quote section of the home page. Again not quite ready for this just yet so only a heads up for now.

TODO @christianliebel

bazzadp commented:
> I also don't work for Google, but I see your point. 

Oops! Sorry I just assumed you did for some reason. Well that in itself makes my point a lot less valid so maybe this isn’t really an issue at all.

> The figures without SQL queries are not drawn from HTTP Archive data, but from the Chrome Platform Status metrics instead. We could add the SQL queries against HTTP Archive for these features as well, but currently, they didn't yield any data. Not sure if Chrome Platform Status metrics have SQL statements as well (or if they should be added to this publication).

Then I think we should add a reference like then instead - like I did in [last year’s HTTP/2 chapter in a couple of places](https://almanac.httparchive.org/en/2019/http2#fig-9). Could do that for all the queries as well to allow readers to see the latest status? Note in that case you add a `caption` (with HTML link) and an `alt` parameter (without any HTML) in the markup:

```
{{ figure_markup(
image="ch20_fig2_http2_usage_by_request.png",
  alt="HTTP/2 usage by request.",
  caption='HTTP/2 usage by request. (Source: <a href="https://httparchive.org/reports/state-of-the-web#h2">HTTP Archive</a>)',
  description="Timeseries chart of HTTP/2 usage showing adoption at 55% for both desktop and mobile as of July 2019. The trend is growing steadily at about 15 points per year.",
  width=600,
  height=321
  )
}}
```

tomayac commented:
> > Where are the queries @tomayac ?
> 
> I added them to the repository in the meantime. All queries can also be found here: https://httparchive.org/reports/project-fugu

Thanks for filling this in for me.

>> I also don't work for Google, but I see your point. The problem there is that, to my knowledge, there's barely any (primary) sources by other vendors with the rare exception of articles like this: https://webkit.org/blog/10855/async-clipboard-api/ But I can definitely go and find some more resources.
>
> TODO @christianliebel / maybe @tomayac is aware of other resources?

Honestly I don't think there are many more resources. There're third party bloggers who wrote about features or the project *per se* ([1](https://medium.com/swlh/what-is-project-fugu-googles-initiative-to-unlock-all-native-device-features-for-the-web-892fafa726f9), [2](https://www.thinktecture.com/en/pwa/fugu/), [3](https://felixgerschau.com/web-capabilities-project-fugu-google/), [4](https://joreteg.com/blog/project-fugu-a-new-hope),…), but not aware of other vendors' publications. There are some relevant [Microsoft Edge Explainers](https://github.com/MicrosoftEdge/MSEdgeExplainers), though, that might be worth linking to.

> The figures without SQL queries are not drawn from HTTP Archive data, but from the Chrome Platform Status metrics instead. We could add the SQL queries against HTTP Archive for these features as well, but currently, they didn't yield any data. Not sure if Chrome Platform Status metrics have SQL statements as well (or if they should be added to this publication).
> 
> @tomayac: What's your opinion?

Chrome Platform Status metrics, as far as I can tell, are only available through indirect means. For example, the stats for [this feature](https://chromestatus.com/metrics/feature/timeline/popularity/2846) come from [that `fetch()` call](https://chromestatus.com/data/timeline/featurepopularity?bucket_id=2846). 

> > > The state of web capabilities in 2020 is healthy, as new powerful APIs regularly ship with new releases of Chromium-based browsers.
> 
> And:
> 
> * Async Clipboard Usage
> * StorageManager API Estimate Usage
> * gIRA() Usage
> 
> WDYT @logicalphase @tomayac ?

I'm biased, but I think it's promisingly healthy, with acknowledged room to grow.

tomayac reviewed (approved):
LGTM, approval

As it was pointed out, my reviews were done in a standalone repo ([final review](https://github.com/christianliebel/web-almanac-2020-capabilities/pull/2), [initial review](https://github.com/christianliebel/web-almanac-2020-capabilities/pull/1)).

obto commented:
What are the todo items left to be taken care of here before we can close this out and add publish it November 9 🎉 ?

christianliebel commented:
> What are the todo items left to be taken care of here before we can close this out and add publish it November 9 🎉 ?

@obto The ones in the first comment. I try to close as many of them as possible until Wednesday at the latest, hope that's ok!

bazzadp commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,313 @@
+reviewers: [logicalphase,tomayac]
+analysts: [tomayac]
+translators: []
+discuss: 1981
```
We have the 2020 discussion ids now and [your own is 2049](https://discuss.httparchive.org/t/chapter-13-capabilities/2049):

```suggestion
discuss: 2049
```

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,313 @@
+reviewers: [logicalphase,tomayac]
+analysts: [tomayac]
+translators: []
+discuss: 2049
```
FYI here's the URL for this discussion thread: https://discuss.httparchive.org/t/chapter-13-capabilities/2049

I'd recommend "watching" it so you can be notified of comments by readers.

bazzadp commented:
Hey @christianliebel we've made some changes that allow you to add the featured quotes and your bio to the meta data at the top of the chapter. See [the example here](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters). Note you are best to merge `main` into your branch before adding this in case of generation issues.

bazzadp commented:
See you're working away at this @christianliebel . Like the MDN references, as they have the added advantage that they are often available in other languages and we update to those where available when translating this chapter. I think between that and the fact you don't actually work at Google as I thought, we've addressed by concerns here so thanks for that.

In fact now gonna suggest the opposite and _adding_ Google links now - I know, I'm fickle me! 😁 Is there a Project Fugu home page that we should link to in the mentions of these in the Introduction? These seem to be the best I can find but will bow to your better knowledge on this as to whether to link to these or better resources.
- https://www.chromium.org/teams/web-capabilities-fugu
- https://web.dev/fugu-status/

Other than that it's looking like we can merge these real soon so let us know when you're done with this round of edits and we'll merge it and iterate for any outstanding items as well as move on to an editing review.

christianliebel commented:
> In fact now gonna suggest the opposite and _adding_ Google links now

Done!

> Other than that it's looking like we can merge these real soon so let us know when you're done with this round of edits and we'll merge it and iterate for any outstanding items as well as move on to an editing review.

I, unfortunately, have to leave for a couple of hours now, I think it'll take ~12 hours until I can finish the rest of the remaining tasks. I'll ping you then.

rviscomi commented:
Left some [feedback](https://docs.google.com/spreadsheets/d/1N6j1qKv7vc51p1W9z_RrbJX9Se3Pmb-Uersfz4gWqqs/edit?disco=AAAAKjGJfCY) in the results sheet about the data viz.

christianliebel commented:
> Other than that it's looking like we can merge these real soon so let us know when you're done with this round of edits and we'll merge it and iterate for any outstanding items as well as move on to an editing review.

@bazzadp Done.

> Left some feedback in the results sheet about the data viz.

@rviscomi Done & replied.

bazzadp commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,335 @@
+
+{{ figure_markup(
+  image="storage_manager_api_estimate.png",
+  alt="Percent of top pages using the estimate method of the StorageManager API"
```
Missing a comma here and in a couple of other places like this, which is causing the test failures. Could you correct?

```suggestion
  alt="Percent of top pages using the estimate method of the StorageManager API",
```

rviscomi replied:
@christianliebel I was testing locally and went ahead and committed these fixes

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+  )
+}}
+
+In April 2020, Google Chrome 81 shipped the new Badging API, followed by Microsoft Edge 84 in July. After Chrome shipped the API, the usage numbers shot up. In October 2020, on 0.025 percent of all page loads in Google Chrome, the `setAppBadge()` method is called. The `clearAppBadge()` method is less often called, during around 0.016 percent of page loads.
```
Absolute page counts

Also, you mentioned that Twitter uses it. It could be insightful to query for all the URLs using it (or browse the sample URLs on the Chrome Status page itself) to find ones that are used on particularly large sites. Do you also see that this API has high usage in the % page views chart? That stat might be useful for comparison to make the point that few sites are using it but the ones that do are very large.

christianliebel replied:
Absolute page counts are not available (page loads), I'll have a look at the rest

christianliebel replied:
The stats use Chrome Platform Status data again (where we neither know the exact sites nor have absolute page counts available). HTTPArchive has only data available for the past three months. Both show a sudden growth in October. I don't see Twitter in the sample URLs though. The sample URLs seem to show various forums. It of course makes sense to present a badge if you decide to add the forum site to your homescreen. How should we proceed here?

rviscomi commented on src/content/en/2020/capabilities.md:
````diff
@@ -0,0 +1,336 @@
+
+The Push API requires the user to be online to receive a notification. Some applications, such as games, reminder or to-do apps, calendars, or alarm clocks, could also determine the target date for a notification locally and schedule it. To support this feature, the Chrome team experiments with a new API called [Notification Triggers](https://web.dev/notification-triggers/) ([Explainer](https://g ...
+
+```js
````
(Side note: @barrypollard it'd be nice to support syntax highlighting)

bazzadp replied:
We do support a limited form and use it in the methodology chapter but it basically needs to be lovely written by hand at the moment:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/a5cbe77943ab7202c9ed55fdef71b20560a62033/src/templates/en/2020/methodology.html#L88-L104

That may be sufficient for now but a nicer code formatter would be nicer alright.

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+  )
+}}
+
+The Fugu team first experimented with Notification Triggers in an origin trial from Chrome 80 to 83, pausing development afterwards due to the lack of feedback by developers. Starting from Chrome 86 released in October 2020, the API has entered the origin trial phase again. This also explains the usage data of Notification Triggers API that peaked at being called on 0.000032 percent of page loads ...
```
Absolute page counts

christianliebel replied:
Not available for Notification Triggers (page loads)

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+
+[Progressive Web Apps](./pwa) (PWA) are a cross-platform application model based on web technology. With the help of Service Workers, these applications run even when the user is offline. The Web App Manifest allows users to add a PWA to their home screen or program list. When opened from there, a PWA appears as a native application. However, PWAs can only use the functions and capabilities that  ...
+
+The [Capabilities Project](https://www.chromium.org/teams/web-capabilities-fugu), informally also known as Project Fugu, is a cross-company effort by Google, Microsoft, and Intel to bridge the gap between web and native. To do so, the Chromium contributors implement new APIs exposing capabilities of the operating system to the web, while maintaining user security, privacy, and trust. These capabi ...
```
Can you talk about why it's important to bridge the gap between web and native? I think the answer has profound implications on the future of the web. Maybe it's too big a question to answer here, but it's something I generally don't see discussed enough. This chapter and Mobile Web are probably the best places to have that discussion, even briefly.

christianliebel replied:
Added a brief sentence. Not sure if this suffices. 😇

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+
+The codename of the Capabilities project is named after a Japanese dish: Correctly prepared, the meat of the blowfish is a special taste experience. If prepared incorrectly, however, it can be fatal. The powerful APIs of Project Fugu are extremely exciting for developers. However, they can affect the security and privacy of the user. Therefore, the Fugu team pays special attention to these issues ...
+
+This chapter gives an overview of various modern web APIs, and the state of web capabilities in 2020 based on usage data by the HTTP Archive. Due to technical limitations, the HTTP Archive only has data available for APIs that require neither permission, nor a user gesture. Where no data is available, the percentage of page loads in Google Chrome according to [Chrome Platform Status](https://chro ...
```
One other point that could be good to make here is that measuring low adoption just means that we have a baseline from which to compare future growth. Future annual editions of this chapter can look back to these stats to see how much the APIs have matured and improved their adoption.

rviscomi commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+
+The state of web capabilities in 2020 is healthy, as new powerful APIs regularly ship with new releases of Chromium-based browsers. Some interfaces like the Content Indexing API or Idle Detection API help to add finishing touches to certain web applications. Other APIs, such as the File System Access and Async Clipboard API, allow a whole new application category, namely productivity apps, to fin ...
+
+Through its [rigorous process](https://developers.google.com/web/updates/capabilities#process), the Fugu team ensures that access to these features takes place in a secure and privacy-friendly manner. Additionally, the Fugu team actively solicits the feedback from other browser vendors and web developers. While the usage of most of these new APIs is comparatively low, some APIs presented in this  ...
```
Could we include a link for the place where developers can give feedback on these APIs?

rviscomi reviewed (changes requested):
@christianliebel this is an excellent chapter and very well-written. Nicely done! 👏 👏 👏 

Most of my feedback is around aligning the HA-based stats with absolute page counts. Any chart changes would also need to have corresponding screenshot updates for the non-interactive versions. Sorry, I know that part can be tedious.

Once this feedback is addressed we can pass it off for a final editing pass before publishing, which I expect to go quickly. I know this has been sitting on my plate for a while so I apologize for getting to it so late. But if you have time to work on it in the next day we can include it in Monday's early access release with the Markup chapter. Or if we need another day I'm sure that can be arranged so both can go out at the same time.

christianliebel commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+  )
+}}
+
+The `persist()` API is less often called than the `estimate()` method. Only 176 mobile pages make use of this API, compared to 25 desktop websites. While usage on the desktop seems to remain at this low level, there is no clear trend on mobile devices.
```
Yes, it would be cool if you could assist me here.

* I've checked [the sample URLs here](https://chromestatus.com/metrics/feature/timeline/popularity/1369). The significantly higher number of mobile sites could be due to the classified ad provider Locanto, which uses the API on different subdomains, all of which count as separate sites.
* Apart from that, using the persist() API on mobile devices probably makes more sense, as they tend to have a smaller storage capacity compared to desktop devices.

christianliebel commented:
> But if you have time to work on it in the next day we can include it in Monday's early access release with the Markup chapter.

@rviscomi Tried to address the feedback and answered your review comments from above, please check them.

bazzadp commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+  image="storage_manager_api_estimate.png",
+  alt="Number of top pages using the estimate method of the StorageManager API",
+  caption="Number of top pages using the estimate method of the StorageManager API.",
+  description="Chart of the usage of StorageManager API's estimate method, based on the percent of pages monitored by HTTPArchive. It compares the usage on mobile and desktop devices. It shows a linear growth on the desktop, while it shows a hockey stick growth for mobile devices. In October, approximately 0.5 percent of both mobile and desktop pages make use of it.",
```
Need to change this now graph is using absolute numbers:

```suggestion
  description="Chart of the usage of StorageManager API's estimate method, based on the number of pages monitored by HTTPArchive. It compares the usage on mobile and desktop devices. It shows a linear growth on the desktop, while it shows a hockey stick growth for mobile devices. In October, approximately 34,000 mobile sites and 27,000 desktop sites make use of it.",
```

bazzadp commented on src/content/en/2020/capabilities.md:
```diff
@@ -0,0 +1,336 @@
+
+{{ figure_markup(
+  image="storage_manager_api_persist.png",
+  alt="Percent of top pages using the persist method of the StorageManager API",
+  caption="Percent of top pages using the persist method of the StorageManager API.",
```
```suggestion
  alt="Number of top pages using the persist method of the StorageManager API",
  caption="Number of top pages using the persist method of the StorageManager API.",
```

bazzadp reviewed (approved):
I think this is good to merge from my perspective

I've made a few minor changes:
- Merged `main` to fix the test error
- Updated some chart descriptions which still referenced % even though the chart was changed. Will just accept these now myself - hope you don't mind.

@rviscomi can you merge away whenever you're happy?

Great job @christianliebel and @tomayac !

christianliebel commented:
> Updated some chart descriptions which still referenced % even though the chart was changed. Will just accept these now myself - hope you don't mind.

@bazzadp Good catch, thanks!

rviscomi commented:
Thanks @christianliebel!

tomayac commented:
Thanks for and congratulations on landing this chapter. This is brilliant work! 👏
